### ISSUE TITLE
[Documentation] Give permalinks more meaningful accessible names

### ISSUE THREAD

**Issue by melanierichards** [WICG/open-ui#176]
While chatting about documentation, we noticed that the permalinks in heading titles are labeled with an `aria-label="Anchor"`. These should be renamed/restructured so that the titles are more meaningful to assistive tech users, e.g. when navigating the content by links.

Example: inspect headings on https://open-ui.org/charter
